packages/vm/src/context.rs

```diff
@@ -55,12 +55,15 @@ pub fn do_read<S: Storage, Q: Querier>(ctx: &Ctx, key_ptr: u32, value_ptr: u32)
         Err(Error::RegionLengthTooBigErr { .. }) => return ERROR_REGION_READ_LENGTH_TOO_BIG,
         Err(_) => return ERROR_REGION_READ_UNKNOWN,
     };
-    let value: Option<Vec<u8>> =
-        match with_storage_from_context::<S, Q, _, _>(ctx, |store| Ok(store.get(&key))) {
-            Ok(v) => v,
-            Err(Error::UninitializedContextData { .. }) => return ERROR_NO_CONTEXT_DATA,
-            Err(_) => return ERROR_DB_UNKNOWN,
-        };
+    let value: Option<Vec<u8>> = match with_storage_from_context::<S, Q, _, _>(ctx, |store| {
+        store
+            .get(&key)
+            .or_else(|_| make_runtime_err("Error reading from backend"))
```

Comment:
Nice.

But, do we want to generalize this (and maintain more info), in that we have a custom `cosmwasm_vm::Error` variant that takes `cosmwasm_std::Error` as the source. This would allow us to maintain more info. (Although we seem to throw it all away in the next lines).

Reply:
I was about to add backend database errors to `cosmwasm_vm::Error` but then we add a lot of overhead for every error type.

> have a custom `cosmwasm_vm::Error` variant that takes `cosmwasm_std::Error` as the source.

In this particular case I think this is misleading, as it somehow implies the error is coming from the contract. Instead, the error comes from the other side (the backend) and just uses the same error type because the `Storage` trait is reused.

I am not 100 % happy with all of this, but concluded that the RuntimeErr is good enough for now.

Reply:
Fair enough. Especially since we never use that error except in the `match` right after